docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/internal/paged-attention.rst

```diff
@@ -0,0 +1,129 @@
+
+---
+
+**Algorithm Explanation**
+
+At the core, attention is computed as:
+
+\[
+\text{Attention}(Q, K, V) = \text{softmax}\left(\frac{QK^T}{\sqrt{d_k}} + \text{Bias}\right)V
+\]
+
+**With enhancements:**
+
+- **KV Cache Layout**: Keys and values are stored in *blocks* of shape `[block_size, head_size]` indexed by `block_indices`. Each sequence may use different blocks for its context.
+- **Subsequence Tokens**: The `subsequence_begins` input defines token start offsets for each sequence in the token batch.
+- **Block Insertion**: When new tokens are processed, their `K` and `V` are inserted into KV cache blocks at locations determined by `block_indices`, updating these blocks in-place.
+- **Optional Eviction Logic**: Controlled externally or internally (if enabled via kernel logic), older blocks may be replaced to maintain memory limits.
+
+**Note:** `block_indices` does not track which blocks are empty or free. It assumes the external memory manager has already assigned valid, available blocks for insertion.
+
+---
+
+**KV Cache Insertion – Pseudocode**
+
+.. code-block:: python
+
+    def insert_into_cache(key_cache, value_cache, key, value, block_indices, block_starts, block_size):
+        """
+        Insert new key/value tensors into cache blocks.
+        Assumes all block_indices have already been allocated and are valid.
+        """
+        token_idx = 0
+        for seq_id, block_begin in enumerate(block_starts[:-1]):
+            seq_token_count = block_starts[seq_id + 1] - block_begin
+            # Blocks assigned to this sequence
+            assigned_blocks = block_indices[block_begin : block_begin + (seq_token_count + block_size - 1) // block_size]
+
+            for i in range(seq_token_count):
+                block_id = assigned_blocks[i // block_size]
+                offset = i % block_size
+                key_cache[block_id, :, offset, :] = key[token_idx]
+                value_cache[block_id, :, offset, :] = value[token_idx]
+                token_idx += 1
+
+---
+
+**Inputs**
+
+1. **query** – `[batch_size_in_tokens, num_heads * head_size]`, type *T*
+2. **key** – `[batch_size_in_tokens, num_kv_heads * head_size]`, type *T*
+3. **value** – `[batch_size_in_tokens, num_kv_heads * head_size]`, type *T*
+4. **key_cache** – `[num_blocks, num_kv_heads, block_size, head_size]`, type *T*
+5. **value_cache** – same shape and type as key_cache
+6. **past_lens** – `[batch_size_in_sequences]`, type `int32`
+7. **subsequence_begins** – `[batch_size_in_sequences + 1]`, type `int32`
+8. **block_indices** – `[num_blocks]`, type `int32`
+9. **block_indices_begins** – `[batch_size_in_sequences + 1]`, type `int32`
+10. **scale** – scalar, type *T* (optional)
+11. **sliding_window** – scalar, type `int32` (optional)
+12. **alibi_slopes** – `[num_kv_heads]`, type *T* (optional)
+13. **rotated_block_indices** – `[num_rotated_blocks]`, type `int32` (optional)
```

Comment:
There is a new input `score_aggregation_window` already added to master.

Reply:
Added